File: src/main/java/org/blacksun/graph/algorithms/BFAlgorithmFactory.java
package org.blacksun.graph.algorithms;

import org.blacksun.graph.channel.Channel;
import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class BFAlgorithmFactory implements PathFindingAlgorithmFactory {
    @Override
    public PathFindingAlgorithm getAlgorithm(@NotNull List<GraphNode> nodes,
                                             @NotNull List<Channel> links) {
        return new BellmanFordAlgorithm(nodes, links);
    }
}

File: src/main/java/org/blacksun/graph/algorithms/BFAlgorithmFactory.kt
package org.blacksun.graph.algorithms

import org.blacksun.graph.channel.Channel
import org.blacksun.graph.node.GraphNode

class BFAlgorithmFactory : PathFindingAlgorithmFactory {
    override fun getAlgorithm(nodes: List<GraphNode>,
                              links: List<Channel>): PathFindingAlgorithm {
        return BellmanFordAlgorithm(nodes, links)
    }
}
